src/main/java/org/variantsync/diffdetective/diff/git/GitDiffer.java

```diff
@@ -360,6 +363,17 @@ private static CommitDiffResult getPatchDiffs(
                         }
                     };
 
+                    // Iff a file does not end with a newline character, git adds a meta-line to the diff, which states
+                    // the absence of the newline. If this is the case, we remove the meta-line, in order not to
+                    // parse it as artifact line. If the meta-line does not exist, we add the newline character.
+                    // TODO: In future versions, we might want to track the newline more explicitly
+                    final Matcher newlineMatcher = NO_NEWLINE_PATTERN.matcher(fullDiff);
+                    if (newlineMatcher.find()) {
+                        fullDiff = fullDiff.substring(0, newlineMatcher.start());
+                    } else {
+                        fullDiff += StringUtils.LINEBREAK;
```

Comment:
This isn't redundant? I though in this case the file ends with a new line character anyway.

Reply:
Hmm... it did not appear in the tree (as seen in the Debugger), and my analysis did not pick it up. I will have a closer look to see what is going on here.

Reply:
We should just make sure to not introduce an extra empty line which will appear as a new DiffNode again.

Reply:
Thank you! :) Would you like to add this finding to your documentation?